src/pages/Main.tsx

```diff
@@ -77,118 +74,109 @@ export const Home: FC = () => {
-    if (!context) return;
-
-    const data = JSON.parse(context?.settings.mapping ?? "{}");
+    const data = getLayout(context?.settings?.mapping);
 
     if (!data) {
       setMappedFields([]);
       setHasMappedFields(false);
     }
+
     setMappedFields(data.listView ?? []);
     setHasMappedFields(!!data.listView?.length);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [context]);
 
   const usableFields = useMemo(() => {
     if (!metadataFieldsQuery.data || hasMappedFields === undefined) return [];
 
-    return metadataFieldsQuery.data.filter((field) =>
-      (hasMappedFields ? mappedFields : IssueJson.main).includes(field.key),
-    );
+    return metadataFieldsQuery.data.filter((field) => {
+      return (hasMappedFields ? mappedFields : IssueJson.main).includes(field.key);
+    });
   }, [metadataFieldsQuery.data, hasMappedFields, mappedFields]);
 
   const debounceTargetAction = useDebouncedCallback<
-    (a: TargetAction<ReplyBoxNoteSelection[]>) => void
+    (a: TargetAction<ReplyBoxSelection[]>) => void
   >((action: TargetAction) => {
-    match<string>(action.name)
-      .with("jiraReplyBoxNoteAdditions", () =>
-        (action.payload ?? []).forEach(
-          (selection: { id: string; selected: boolean }) => {
-            const ticketId = action.subject;
-
-            if (context?.data.ticket.id) {
-              client
-                ?.setState(
-                  ticketReplyNotesSelectionStateKey(ticketId, selection.id),
-                  { id: selection.id, selected: selection.selected },
-                )
-                .then((result) => {
-                  if (result.isSuccess) {
-                    registerReplyBoxNotesAdditionsTargetAction(
-                      client,
-                      context.data.ticket.id,
-                      linkedIssues,
-                    );
-                  }
-                });
-            }
-          },
-        ),
-      )
-      .with("jiraReplyBoxEmailAdditions", () =>
-        (action.payload ?? []).forEach(
-          (selection: { id: string; selected: boolean }) => {
-            const ticketId = action.subject;
-
-            if (context?.data.ticket.id) {
-              client
-                ?.setState(
-                  ticketReplyEmailsSelectionStateKey(ticketId, selection.id),
-                  { id: selection.id, selected: selection.selected },
-                )
-                .then((result) => {
-                  if (result.isSuccess) {
-                    registerReplyBoxEmailsAdditionsTargetAction(
-                      client,
-                      context.data.ticket.id,
-                      linkedIssues,
-                    );
-                  }
-                });
-            }
-          },
-        ),
-      )
+    match(action.name)
+      .with("jiraReplyBoxNoteAdditions", () => {
+        ((action as { payload: ReplyBoxSelection[] }).payload ?? []).forEach((selection: ReplyBoxSelection) => {
```

Comment:
We shouldn't need to use `as` here if `action` is properly typed. Can we make `action` be stronger typed so we can remove `as`?